File: solutions/HJK9810/2026-09-01/low/27025.java
import java.io.BufferedReader;
import java.io.InputStreamReader;

class Solution {
	public static void main(String args[]) throws Exception {
		BufferedReader input = new BufferedReader(new InputStreamReader(System.in));
    StringBuilder sb = new StringBuilder();
    int T = Integer.parseInt(input.readLine());

		for(int test_case = 1; test_case <= T; test_case++) {
      int N = Integer.parseInt(input.readLine());
      String[] members = input.readLine().split(" ");
      int K = Integer.parseInt(input.readLine());
      String[] canceledMembers = input.readLine().split(" ");

      for (int idx = 0; idx < N; idx++) {
        for (int index = 0; index < K; index++) {
          if (members[idx].equals(canceledMembers[index])) {
            members[idx] = "";
            canceledMembers[index] = "";
            break;
          }
        }
      }

      boolean hasMember = false;
      sb.append("#" + test_case);
      for (int index = 0; index < N; index++) {
        if (members[index].isEmpty()) continue;
        hasMember = true;
        sb.append(" " + members[index]);
      }

      if (!hasMember) sb.append(" empty");
      sb.append("\n");
		}
    
    System.out.print(sb.toString());
	}
}

File: solutions/HJK9810/2026-09-01/low/27025.py
def algorithm():
    N = int(input())
    numbers = list(map(int, input().split()))
    K = int(input())
    cancels = list(map(int, input().split()))

    for idx in range(N):
        for index in range(K):
            if numbers[idx] == cancels[index]:
                numbers[idx] = 0
                cancels[index] = 0
                break

    members = [str(num) for num in numbers if num != 0]
    return " ".join(members) if len(members) > 0 else "empty"

T = int(input())
# 여러개의 테스트 케이스가 주어지므로, 각각을 처리합니다.
for test_case in range(1, T + 1):
    print(f"#{test_case} {algorithm()}")
